1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,25 +1,22 @@
 function setAlarm() {
   const setAlarmButton = document.querySelector("#set");
   let displayTime = document.querySelector("#second");
-
+  document.body.style.backgroundColor = "lightblue";
   setAlarmButton.addEventListener("click", function () {
-    const inputValue = document.querySelector("#alarmSet").value;
-    displayTime.innerHTML = inputValue;
-
-    setInterval(() => {
-      if (displayTime.innerHTML > 0) {
-       document.body.style.backgroundColor = "green";
-        displayTime.innerHTML--;
-        
-      } else {
-        displayTime = 0;
-        document.body.style.backgroundColor = "orange";
-      }
-    }, 1000);
+    let seconds = document.querySelector("#alarmSet").value;
 
+    const countdown = setInterval(() => {
+      seconds--;
+      displayTime.innerHTML = seconds;
+      console.log(seconds);
 
+      if (seconds === 0) {
+        playAlarm();
+        document.body.style.backgroundColor = "lightgreen";
+        clearInterval(countdown);
```

Comment:
Spot on, simple, elegant solution to the problem. I particuluarly liked that you rembered to clear the interval when you're done, otherwise you'd have multiple intervals confusing themselves